content/300-guides/100-performance-and-optimization/150-connection-management/index.mdx

```diff
@@ -0,0 +1,286 @@
+
+The following example demonstrates how to set the `connection_limit` to 1 in your connection string:
+
+<TabbedContent tabs={[<FileWithIcon text="PostgreSQL" icon="database"/>, <FileWithIcon text="MySQL" icon="database"/>]}>
+<tab>
+
+  ```
+  postgresql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
+  ```
+
+</tab>
+<tab>
+
+  ```
+  mysql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
+  ```
+
+</tab>
+</TabbedContent>
+
+#### With an external connection pooler
+
+If you are using an external connection pooler, use the default connection limit (`num_physical_cpus * 2 + 1`) as a starting point and then [tune the pool size](#optimizing-the-connection-pool). The external connection pooler prevents a traffic spike from overwhelming the database.
+
+### <inlinecode>PrismaClient</inlinecode> in serverless environments
+
+#### Instantiate `PrismaClient` outside the handler
+
+Instantiate `PrismaClient` [outside the scope of the function handler](https://github.com/prisma/e2e-tests/blob/5d1041d3f19245d3d237d959eca94d1d796e3a52/platforms/serverless-lambda/index.ts#L3) to increase the chances of reuse. As long as the handler remains 'warm' (in use), the connection is potentially reusable:
+
+```ts highlight=3;normal
+import { PrismaClient, Prisma } from '@prisma/client'
+
+const client = new PrismaClient()
+
+export async function handler() {
+    /* ... */
+}
+```
+
+#### Do not explicitly <inlinecode>$disconnect()</inlinecode>
+
+You [do not need to explicitly `$disconnect()`](../../../concepts/components/prisma-client/working-with-prismaclient/connection-management#calling-disconnect-explicitly) <span class="concept"></span> at the end of a function, as there is a possibility that the container might be reused. Opening a new connection takes time and slows down your function's ability to process requests.
+
+### Other serverless considerations
+
+#### Container reuse
+
+There is no guarnatee that subsequent nearby invocations of a function will hit the same container - for example, AWS can choose to create a new container at any time.
+
+Code should assume the container to be stateless and create a connection only if it does not exist - Prisma Client JS already implements this logic.
+
+#### Zombie connections
+
+Containers that are marked to be removed and are not being reused still keep a connection open and can stay in that state for some time (unknown and not documented from AWS). This can lead to sub-optimal utilization of the DB connections.
+
+Potential solutions<sup>1, 2</sup> include:
+
+* Using a lower idle connection timeout.
+* Cleaning up the idle connections in a separate service.
+
+<sup>
+  1. Note that these are recommendations and not best practices. These would vary from system to
+  system.
+</sup>
+<br />
+<sup>
+  2.{' '}
+  <a href="https://github.com/jeremydaly/serverless-mysql">
+    <inlineCode>serverless-mysql</inlineCode>
+  </a>{' '}
+  is a library that implements this idea.
```

Comment:
It only implements the second idea.

Reply:
Updated: https://607d3769e786c80007ec9d89--prisma2-docs.netlify.app/docs/guides/performance-and-optimization/connection-management#zombie-connections